AccountingService/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using AccountingService.Controllers.models;
+using AccountingService.Data;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using System.Security.Cryptography;
+using BC = BCrypt.Net.BCrypt;
+using System.Text.RegularExpressions;
+
+namespace AccountingService.Controllers
+{
+    [EnableCors("CorsOriginPolicy")]
+    [Route("api/authentication")]
+    [ApiController]
+    public class AuthenticationController : ControllerBase
+    {
+        private SalaryConversionContext salaryConversionContext;
+        public AuthenticationController(SalaryConversionContext context) : base()
+        {
+            salaryConversionContext = context;
+        }
+            [HttpPost("register")]
+        public async Task<ActionResult<SystemUser>> Register(Authentication requst)
+        {
+            CreatePasswordHash(requst.Password, out byte[] passwordHash, out byte[] passwordSalt);
+            var user = new SystemUser();
+            try
+            {
+                
+                var useremail= salaryConversionContext.SystemUser.
+                    Where(u => u.Email == requst.Email).SingleOrDefault();
+                if (useremail != null)
+                {
+                    return BadRequest("Email already exists.");
+                }
+
+                if (!Regex.IsMatch(requst.Password, @"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\da-zA-Z]).{8,}$"))
```

Comment:
This regex value is a magic string. Move it to a constant and give it a proper name.